itests/fevm_test.go

```diff
@@ -119,22 +259,307 @@ func TestFEVMDelegateCall(t *testing.T) {
+	fmt.Println("ret - ", ret)
+}
+
+// creates a contract that would fail when tx are sent to it
+// on eth but on fevm it succeeds
+// example on goerli of tx failing https://goerli.etherscan.io/address/0xec037bdc9a79420985a53a49fdae3ccf8989909b
+func TestFEVMSendGasLimit(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract
+	filenameActor := "contracts/GasLimitSend.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameActor)
+
+	//send $ to contract
+	//transfer 1 attoFIL to contract
+	sendAmount := big.MustFromString("1")
+
+	client.EVM().TransferValueOrFail(ctx, fromAddr, contractAddr, sendAmount)
+	ret, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "getDataLength()", []byte{})
+	require.NoError(t, err)
+	fmt.Println("ret - ", ret)
+
+}
+
+// TestFEVMDelegateCall deploys the two contracts in TestFEVMDelegateCall but instead of A calling B, A calls A which should cause A to cause A in an infinite loop and should give a reasonable error
+// XXX should not be fatal errors
+func TestFEVMDelegateCallRecursiveFail(t *testing.T) {
+	//TODO change the gas limit of this invocation and confirm that the number of errors is different
+	//also TODO should we not have fatal error show up here?
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameActor := "contracts/DelegatecallStorage.hex"
+	fromAddr, actorAddr := client.EVM().DeployContractFromFilename(ctx, filenameActor)
+
+	//any data will do for this test that fails
+	inputDataContract := inputDataFromFrom(ctx, t, client, actorAddr)
+	inputDataValue := inputDataFromArray([]byte{7})
+	inputData := append(inputDataContract, inputDataValue...)
+
+	//verify that the returned value of the call to setvars is 7
+	_, wait, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "setVarsSelf(address,uint256)", inputData)
+	require.Error(t, err)
+	require.Equal(t, exitcode.SysErrorIllegalArgument, wait.Receipt.ExitCode)
+
+	//assert no fatal errors but still there are errors::
+	error1 := "f01002 (method 2) -- fatal error (10)" // showing once
+	error2 := "f01002 (method 5) -- fatal error (10)" // showing 256 times
+	error3 := "f01002 (method 3) -- fatal error (10)" // showing once
+	errorAny := "fatal error"                         // showing once
+
+	require.NotContains(t, err.Error(), error1)
+	require.NotContains(t, err.Error(), error2)
+	require.NotContains(t, err.Error(), error3)
+	require.NotContains(t, err.Error(), errorAny)
+}
+
+// XXX Currently fails as self destruct has a bug
+// TestFEVMTestSendValueThroughContracts creates A and B contract and exchanges value
+// and self destructs and accounts for value sent
+func TestFEVMTestSendValueThroughContractsAndDestroy(t *testing.T) {
+
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	fromAddr := client.DefaultKey.Address
+	t.Log("from - ", fromAddr)
```

Comment:
I guess this test isn't passing yet, so maybe leave this here for now, but add a //Removeme comment

Reply:
i think this one is ok